src/test/java/dasniko/testcontainers/keycloak/KeycloakContainerTest.java

```diff
@@ -31,6 +34,21 @@ public void shouldStartKeycloak() {
         }
     }
 
+    @Test
+    public void shouldConsiderConfiguredStartupTimeout() {
+        Instant start = Instant.now();
+        try {
+            Duration duration = Duration.ofSeconds(5);
+            try (KeycloakContainer keycloak = new KeycloakContainer().withStartupTimeout(duration)) {
+                keycloak.start();
+            }
+        } catch(ContainerLaunchException ex) {
+            Duration observedDuration = Duration.between(start, Instant.now());
+            assertTrue("Startup time should consider configured limit",
+                observedDuration.toMillis()/1000 > 5 && observedDuration.toMillis()/1000 < 15);
```

Comment:
Hopefully Keycloak will stay so slow during startup that we can rely on this fixed number 😉